src/test/java/no/unit/nva/model/DoesNotHaveNullFields.java

```diff
@@ -0,0 +1,88 @@
+package no.unit.nva.model;
+
+import static java.util.Objects.isNull;
+import static nva.commons.utils.attempt.Try.attempt;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+
+public class DoesNotHaveNullFields<T> extends BaseMatcher<T> {
+
+    public static final String GETTER_GET_PREFIX = "get";
+    public static final String GETTER_IS_PREFIX = "is";
+
+    @Override
+    public boolean matches(Object actual) {
+        return assertThatNoPublicFieldIsNull(actual);
+    }
+
+    @Override
+    public void describeTo(Description description) {
+        description.appendText("All fields to be non empty");
+    }
+
+    public static <R> DoesNotHaveNullFields<R> doesNotHaveNullFields() {
+        return new DoesNotHaveNullFields<>();
+    }
+
+    private boolean assertThatNoPublicFieldIsNull(Object insertedUser) {
+        Method[] methods = insertedUser.getClass().getMethods();
+        Stream<MethodInvocationResult> getterInvocations = Arrays.stream(methods)
+            .filter(this::isAGetter)
+            .map(attempt(method -> invokeMethod(insertedUser, method)))
+            .map(eff -> eff.orElseThrow(fail -> new RuntimeException(fail.getException())));
+
+        List<MethodInvocationResult> emptyArgs = getterInvocations.filter(this::isEmpty).collect(
+            Collectors.toList());
+        return emptyArgs.isEmpty();
+    }
+
+    private MethodInvocationResult invokeMethod(Object insertedUser, Method method)
+        throws IllegalAccessException, InvocationTargetException {
+        Object result = method.invoke(insertedUser);
+        return new MethodInvocationResult(method.getName(), result);
+    }
+
+    private boolean isAGetter(Method m) {
```

Comment:
Compare with 
`

    private List<Method> getters;

    private boolean isAGetter(Method m) {
        return getters.contains(m);
    }

    public static List<Method> findGetters(Object object) throws IntrospectionException {
        return Arrays.stream(Introspector.getBeanInfo(object.getClass(), Object.class)
                .getPropertyDescriptors())
                .filter(DoesNotHaveNullFields::isGetter)
                .map(attempt((propertyDescriptor) -> getGetter(propertyDescriptor, object)))
                .map(effort -> effort.orElseThrow(DoesNotHaveNullFields::unexpectedException))
                .collect(Collectors.toList());
    }

    private static RuntimeException unexpectedException(Failure<Method> failure) {
        throw new IllegalStateException("Something went wrong", failure.getException());
    }

    private static Method getGetter(PropertyDescriptor propertyDescriptor, Object object) {
        try {
            if (nonNull(propertyDescriptor.getReadMethod().invoke(object))) {
                return propertyDescriptor.getReadMethod();
            }
        } catch (IllegalAccessException | InvocationTargetException e) {
            // I guess we should do something here
        }
        // Not this either
        return null;
    }

    private static boolean isGetter(PropertyDescriptor propertyDescriptor) {
        return nonNull(propertyDescriptor.getReadMethod());
    }
`

Reply:
check https://github.com/BIBSYSDEV/nva-testutils/pull/8/files